numpy/core/src/multiarray/getset.c

```diff
@@ -513,6 +516,7 @@ array_descr_set(PyArrayObject *self, PyObject *arg, void *NPY_UNUSED(ignored))
         /* resize on last axis only */
         int axis = PyArray_NDIM(self) - 1;
         if (PyArray_DIMS(self)[axis] != 1 &&
+                PyArray_SIZE(self) != 0 &&
```

Comment:
Hah, this should work, yeah.  So the one thing is that if the stride was previously 0, in theory it should remain 0 in this case (but gets replaced with `elsize`).  I think I am OK with ignoring the potential UB problem that may follow that for now.

If we really want to fight it, I guess we would have to add asserts into `array_dealloc` to flush out places that need adjustment.

@eric-wieser how worried are you actually about the UB?  If not very, I am tempted to suggest accepting that this path might lead to this UB.

Reply:
@mattip if you really care about the prod, you cold order it to the end, since it will be hit very rarely then?
But honestly, this is still a Python "call", so I really doubt it matters at all.